nncf/quantization/algorithms/accuracy_control/evaluator.py

```diff
@@ -0,0 +1,152 @@
+        """
+        Validates model.
+
+        :param model: Model to validate.
+        :param dataset: Dataset to validate the model.
+        :param indices: Zero-based indices of data items that should be selected from
+            the dataset.
+        :return: A tuple (metric_value, values_for_each_item) where
+            - metric_values: This is a metric for the model.
+            - values_for_each_item: If the `Evaluator.is_metric_mode()` condition is true,
+                then `values_for_each_item` represents the list of metric value for each item.
+                Otherwise, if the condition is false, it represents list of logits for each
+                item.
+        """
+        model_for_inference = self._algo_backend.prepare_for_inference(model)
+        if self._metric_mode is None:
+            self._determine_mode(model_for_inference, dataset)
+
+        if not self.is_metric_mode() and indices is not None:
+            raise ValueError("The `indices` parameter can be used only if Evaluator.is_metric_mode() = True")
+
+        metric, values_for_each_item = self._validation_fn(model_for_inference, dataset.get_data(indices))
+
+        return metric, values_for_each_item
+
+    def _determine_mode(self, model_for_inference: TModel, dataset: Dataset) -> None:
+        """
+        Determines mode based on the type of returned value from the
+        validation function.
+
+        :param model_for_inference: Model to validate.
+        :param dataset: Dataset to validate the model.
+        """
+        data_item = dataset.get_data([0])
+        # pylint: disable=W0703
+        try:
+            metric_value, values_for_each_item = self._validation_fn(model_for_inference, data_item)
+        except Exception:
+            self._metric_mode = False
+
+        if self._metric_mode is not None:
+            return
+
+        # Analyze `metric_value` and `values_for_each_item` values:
+        # +--------------+----------------------+-------------+
+        # | metric_value | values_for_each_item | metric_mode |
+        # +--------------+----------------------+-------------+
+        # | float        | None                 | True        |
+        # +--------------+----------------------+-------------+
+        # | float        | List[float]          | True        |
+        # +--------------+----------------------+-------------+
+        # | float        | List[List[TTensor]]  | False       |
+        # +--------------+----------------------+-------------+
+        # | None         | None                 | False       |
+        # +--------------+----------------------+-------------+
+        # | None         | List[float]          | UNEXPECTED  |
+        # +--------------+----------------------+-------------+
+        # | None         | List[List[TTensor]]  | False       |
+        # +--------------+----------------------+-------------+
+
+        self._metric_mode = False
+        if isinstance(metric_value, float) and (
```

Comment:
We talked about metric_value should cast to float. I have checked the isinstance with numpy array and it does not work.

Reply:
Thanks! I fixed it.